expat/configure.ac

```diff
@@ -183,33 +183,75 @@ AC_LINK_IFELSE([AC_LANG_SOURCE([
       AC_MSG_RESULT([yes])],
      [AC_MSG_RESULT([no])])])
 
-
-AC_MSG_CHECKING([for getrandom (Linux 3.17+, glibc 2.25+)])
-AC_LINK_IFELSE([AC_LANG_SOURCE([
-  #include <stdlib.h>  /* for NULL */
-  #include <sys/random.h>
-  int main() {
-    return getrandom(NULL, 0U, 0U);
-  }
-  ])],
-  [AC_DEFINE([HAVE_GETRANDOM], [1], [Define to 1 if you have the `getrandom' function.])
-   AC_MSG_RESULT([yes])],
-  [AC_MSG_RESULT([no])
-
-   AC_MSG_CHECKING([for syscall SYS_getrandom (Linux 3.17+)])
+AC_ARG_WITH([usegetrandom],
+ [AS_HELP_STRING([--with-usegetrandom],
+ [Use getrandom function in the system @<:@default=check@:>@])],
+ [],
+ [with_usegetrandom=check])
+          
+AS_IF([test "x$with_usegetrandom" != xno],
+  [AC_MSG_CHECKING([for getrandom (Linux 3.17+, glibc 2.25+)])
+   AC_LINK_IFELSE([AC_LANG_SOURCE([
+   #include <stdlib.h>  /* for NULL */
+   #include <sys/random.h>
+   int main() {
+     return getrandom(NULL, 0U, 0U);
+   }
+   ])],
+   [
+	AC_DEFINE([HAVE_GETRANDOM], [1], [Define to 1 if you have the `getrandom' function.])
+    AC_MSG_RESULT([yes])
+    getrandom_detect=yes
+   ],
+    [AC_MSG_RESULT([no])]
+     getrandom_detect=no)
+	],
+   [if test "x$with_usegetrandom" != xcheck; then
+       AC_DEFINE([HAVE_GETRANDOM], [0], [Define to 0, dont use the `getrandom' function.])
+       fi
+   ]) 
+
+if test "x$with_usegetrandom" = xyes; then
+   if test "x$getrandom_detect" = xno; then
+    AC_MSG_FAILURE(enforced the use of getrandom with --with-usegetrandom but not detected)
+fi
+fi
+
+AC_ARG_WITH([usesysgetrandom],
+ [AS_HELP_STRING([--with-usesysgetrandom],
+ [Use getrandom function in the system @<:@default=check@:>@])],
+ [],
+ [with_usesysgetrandom=check])
+
+AS_IF([test "x$with_usesysgetrandom" != xno],
+  [AC_MSG_CHECKING([for syscall SYS_getrandom (Linux 3.17+)])
    AC_LINK_IFELSE([AC_LANG_SOURCE([
       #include <stdlib.h>  /* for NULL */
       #include <unistd.h>  /* for syscall */
       #include <sys/syscall.h>  /* for SYS_getrandom */
       int main() {
         syscall(SYS_getrandom, NULL, 0, 0);
         return 0;
-      }
+   }
    ])],
-   [AC_DEFINE([HAVE_SYSCALL_GETRANDOM], [1], [Define to 1 if you have `syscall' and `SYS_getrandom'.])
-    AC_MSG_RESULT([yes])],
-   [AC_MSG_RESULT([no])])])
-
+   [
+        AC_DEFINE([HAVE_SYSCALL_GETRANDOM], [1], [Define to 1 if you have 'syscall' and 'SYS_getrandom'.])
+    AC_MSG_RESULT([yes])
+    sysgetrandom_detect=yes
+   ],
+    [AC_MSG_RESULT([no])]
+     sysgetrandom_detect=no)
+        ],
+   [if test "x$with_usesysgetrandom" != xcheck; then
+       AC_DEFINE([HAVE_SYSCALL_GETRANDOM], [0], [Define to 0, dont use the syscall and SYS_getrandom functions.])
+       fi
+   ])
+
+if test "x$with_usesysgetrandom" = xyes; then
+   if test "x$sysgetrandom_detect" = xno; then
+    AC_MSG_FAILURE(enforced the use of sysgetrandom with --with-usesysgetrandom but not detected)
+fi
+fi
```

Comment:
All (or close to all?) comments above apply to this code again. If you want, we can finish one half and then copy over so you don't need to do more things twice than necessary — as you like.